api/envoy/extensions/http/stateful_session/cookie/v3/cookie.proto

```diff
@@ -0,0 +1,56 @@
+syntax = "proto3";
+
+package envoy.extensions.http.stateful_session.cookie.v3;
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.http.stateful_session.cookie.v3";
+option java_outer_classname = "CookieProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/http/stateful_session/cookie/v3;cookiev3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Cookie based stateful session extension]
+
+// This extension allows the session state to be tracked via cookies.
+//
+// This extension first encodes the address of the upstream endpoint selected by the load
+// balancer into a `set-cookie` response header with the :ref:`configured name
+// <envoy_v3_api_field_extensions.http.stateful_session.cookie.v3.CookieBasedSessionState.name>`.
+// This extension will also try to parse the specific upstream endpoint address from
+// the named cookie when new requests come. If the address parsed from the cookie
+// corresponds to a valid upstream endpoint, this upstream host will be selected first.
+// See :ref:`stateful session filter
+// <envoy_v3_api_msg_extensions.filters.http.stateful_session.v3.StatefulSession>`.
+//
+// For example, if the cookie name is set to `sticky-host`, envoy will prefer `1.2.3.4:80`
+// as the upstream endpoint when the request contains the following header:
+//
+// .. code-block:: none
+//
+//     cookie: sticky-host="MS4yLjMuNDo4MA=="
+//
+// When processing the upstream response, if endpoint `1.2.3.4:80` is indeed the final choice
+// (endpoint `1.2.3.4:80` exists and its status is as expected), the extension does nothing.
+// If endpoint `1.2.3.4:80` is not the final choice, the new selected host will be used to
+// update the cookie (via the `set-cookie` response header).
+//
+// [#extension: envoy.http.stateful_session.cookie]
+message CookieBasedSessionState {
+  // The name of the cookie that will be used to store the encoded upstream host.
+  // The value of the cookie is the upstream host address string encoded with base64.
+  // If in the request process, it will pass through multiple Envoy proxies. And multiple
+  // Envoy proxies use this feature. Please keep the name different so as not to confuse
+  // the content of the cookie.
+  string name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // If specified, a TTL will be added to new session state cookies in the set-cookie header.
+  // Set this to 0 to use a session cookie.
+  google.protobuf.Duration ttl = 2 [(validate.rules).duration = {gte {}}];
+
+  // The path for the cookie. If no path is specified here, no path will be set for the cookie.
+  string path = 3;
```

Comment:
Why can't we reuse the `Cookie` from `HashPolicy` in `route_components.proto`?

Reply:
But `Cookie` is part of `HashPolicy`, so a separate `CookieBasedSessionState` is created.
If we use `Cookie` in `HashPolicy`, I am worried that users will misunderstand this function and the meaning of the API.
If `Cookie` can be moved to a more public part, I think it is entirely possible for us to take it directly.

Reply:
We can't move it at this point in v3. That said, I would rather we didn't have to do this again next time someone wants a cookie description. Can you add a generic Cookie type to https://github.com/envoyproxy/envoy/tree/main/api/envoy/type/http/v3?

Reply:
get it.